bin/exec.go

```diff
@@ -85,93 +77,49 @@ func main() {
 
 		// If not found in AppNamespace skip the further steps
 		if !isFound {
+			//TODO Patch Engine if chaosExperiment is not found.
+			engineDetails.ExperimentNotFoundPatchEngine(experiment, clients)
+
 			log.Infoln("Can't Find Experiment Name : "+engineDetails.Experiments[i], "In Namespace : "+engineDetails.AppNamespace)
 			log.Infoln("Not Executing the Experiment : " + engineDetails.Experiments[i])
 			break
 		}
+		log.Info("Printing the ENV Variables", experiment.Env)
 
-		log.Infoln("Getting the Default ENV Variables")
-
-		// Get the Default ENV's from ChaosExperiment
-		experiment.SetDefaultEnv(clients)
-
-		log.Info("Printing the Default Variables", experiment.Env)
-
-		// Get the ConfigMaps for patching them in the job creation
-		log.Infoln("Find the configMaps in the chaosExperiments")
-
-		experiment.SetConfigMaps(clients)
-
-		err := experiment.ValidateConfigMaps(clients)
+		// Patch ConfigMaps to ChaosExperiment Job
+		err := experiment.PatchConfigMaps(clients)
 		if err != nil {
-			log.Infof("Aborting Execution")
-			return
+			break
 		}
-
-		experiment.SetSecrets(clients)
-		err = experiment.ValidateSecrets(clients)
+		// Patch Secrets to ChaosExperiment Job
+		err = experiment.PatchSecrets(clients)
 		if err != nil {
-			log.Infof("Aborting Execution")
-			return
+			break
 		}
-
-		log.Infof("Validated ConfigMaps: %v", experiment.ConfigMaps)
-		log.Infof("Validated Secrets: %v", experiment.Secrets)
-
 		// Adding VolumeBuilders, according to the ConfigMaps, and Secrets from ChaosEXperiment
 		experiment.VolumeOpts.VolumeBuilders = utils.CreateVolumeBuilder(experiment.ConfigMaps, experiment.Secrets)
```

Comment:
##TODO: Each volume resource should have it's own volumeBuilder function

Reply:
Done.